### PR TITLE
use systematic way to locate third-party lib

### LINKED ISSUE #280: The way of finding PUI needs to be improved
The way of finding PUI in [`modmesh/view.py`](https://github.com/solvcon/modmesh/blob/master/modmesh/view.py#L67) is using hard-code import path, which is not a good approach.
This is a point that needs improvement.

### PR DESCRIPTION
**Pull request by j8xixo12**
I added a python script in `thirdparty` folder that provides `register_library` and `load_library` for modmesh to select which  library to load.

### CODE REVIEW

File: modmesh/core.py
Comment (yungyuc) [inline modmesh/core.py]: It is harder to read the lines when they are broken like:

```python
if path is None or not os.path.exists(os.path.join(path,
                                                   filename)):
```

In this case, breaking the line using a local variable is more readable:

```python
_pf = os.path.join(path, filename)
if path is None or not os.path.exists(_pf):
```

File: modmesh/core.py
Comment (yungyuc) [inline modmesh/core.py]: Do you need the trailing `\n`?

File: modmesh/core.py
Comment (yungyuc) [inline modmesh/core.py]: You need a unit test for the loop so that when it goes wrong and into infinite, there is a place to add regression.

File: modmesh/view.py
Comment (yungyuc) [inline modmesh/view.py]: It is very hard to know where to find `python_lib_management`.  A maintainer needs to find everywhere in the source package.  It will be hopeless when `modmesh` is deployed to system.

If you use it this way, I think the module needs to be placed under `modmesh` namespace.  If it's placed under `modmesh`, the module needs a better (more precise) name.

File: modmesh/core.py
Comment (yungyuc) [inline modmesh/core.py]: The whole logic should be placed in a function and documented in the docstring.  It's hard to understand.

File: thirdparty/python_lib_management.py
Comment (yungyuc) [inline thirdparty/python_lib_management.py:1]: You are not making this file a script but a module, and make it unreasonable to place the file under `thirdparty/`.

**Review by yungyuc — CHANGES_REQUESTED** [review]
- [x] Make `thirdparty/` path searching testable and add the docstrings and tests.
- [x] Beautify line breaking.
- [x] Discuss where to place the module `python_lib_management` and its name.

**Comment by yungyuc** [comment]
This PR is for #280 .  @j8xixo12 , next time, please note the association to the issue (write "ref xxx" or "for xxx", not "resolve" or "address" xxx which may close the associated issue.)

File: modmesh/view.py
Comment (j8xixo12) [reply modmesh/view.py]: I will move the `python_lib_management` from `thirdparty` to `modmesh`.
How about rename it to `libutil` ?

File: modmesh/core.py
Comment (j8xixo12) [reply modmesh/core.py]: Yes, without tailing `\n`, the message on the screen will become:
```
Can not find thirdparty.No python 3rd-party library will be imported
```

File: modmesh/view.py
Comment (yungyuc) [reply modmesh/view.py]: I would like to have more descriptive name.  Let's try `pylibmgr`?

File: modmesh/core.py
Comment (yungyuc) [reply modmesh/core.py]: I meant the `\n` in `... imported\n`.

Comment (j8xixo12) [reply modmesh/core.py]: Oh, it still needs `\n` after `....import`, without `\n`, the shell prompt will not start on a  newline:
```
Can not find thirdparty.
No python 3rd-party library will be imported$
```
it will not look not good.

File: modmesh/view.py
Comment (j8xixo12) [reply modmesh/view.py]: LGTM, let's try `pylibmgr`!

File: modmesh/core.py
Comment (yungyuc) [reply modmesh/core.py]: I see.  Thanks.

File: modmesh/pylibmgr.py
Comment (j8xixo12) [inline modmesh/pylibmgr.py:47]: Use a timeout here to ensure `modmesh` does not get stuck in an infinite loop.

File: modmesh/pylibmgr.py
Comment (j8xixo12) [inline modmesh/pylibmgr.py:53]: Use a timeout here to ensure `modmesh` does not stuck in an infinite loop.

File: modmesh/core.py
Comment (yungyuc) [inline modmesh/core.py]: Use `from . import pylibmgr` and `pylibmgr.search_library_root()` later.  It will be easier to find module use outside this file during maintenance.

File: modmesh/pylibmgr.py
Comment (yungyuc) [inline modmesh/pylibmgr.py:39]: Add the docstring for `timeout`:

```
:ivar timeout: Maximum time for root path searching.  Default is 1.0 second.
```

File: modmesh/pylibmgr.py
Comment (yungyuc) [inline modmesh/pylibmgr.py]: Since `timeout` is in second, it's clearer to write the default value as a float.

File: modmesh/pylibmgr.py
Comment (yungyuc) [reply modmesh/pylibmgr.py:53]: This is nice.

File: modmesh/pylibmgr.py
Comment (yungyuc) [inline modmesh/pylibmgr.py:82]: Appending `sys.path` deteriorate maintainability because it is hard to find this line and the value input to `append()` later during maintenance.  But I agree this is a straight-forward way to make the thirdparty importable.

If we have to do this, please create an issue to track for the improvement right before this is to be merged, and add a code comment to point to the issue.

File: modmesh/view.py
Comment (yungyuc) [inline modmesh/view.py]: Ditto. Use module import.

File: tests/test_pylibmgr.py
Comment (yungyuc) [inline tests/test_pylibmgr.py:39]: Please add a code comment stating that this line assumes the test directory is in the source package, in which the `modmesh` Python directory exists.

**Review by yungyuc — CHANGES_REQUESTED** [review]
- [x] Use module import and access functions inside, instead of import the function.
- [x] Use float for `timeout` argument default and add docstring.
- [x] Deal with the difficulty of maintenance with `sys.path` entry insertion.

File: tests/test_pylibmgr.py
Comment (j8xixo12) [reply tests/test_pylibmgr.py:39]: I'm thinking this test case assume that the third-party library root folder name is `thirdparty`. If the `thirdparty` is renamed, we will need to modify this test case.
If I create a folder in `tests` by `os.mkdir` in `setUp`, this folder can be used for `test_pylibmgr_search_library_root` and this folder will be removed in `tearDown`.

I'm not sure if it's appropriate that create / remove folder in unit-test code ?

Comment (yungyuc) [reply tests/test_pylibmgr.py:39]: I don't think we need anything more than a code comment at the moment.

File: modmesh/pylibmgr.py
Comment (j8xixo12) [reply modmesh/pylibmgr.py:82]: I have created an issue https://github.com/solvcon/modmesh/issues/378 to track the improvement for importing python third-party library.

**Comment by j8xixo12** [comment]
- [x] Use module import and access functions inside, instead of import the function.
- [x] Use float for `timeout` argument default and add docstring.
- [x] Deal with the difficulty of maintenance with `sys.path` entry insertion.

@yungyuc This PR is ready to review!

**Comment by yungyuc** [comment]
LGTM.  Merging.

**Comment by yungyuc** [comment]
ref #280

**Pull request merged**
